### PR TITLE
Caching#54

### PR DESCRIPTION
**Pull request by asmolentzov**
Implements some caching in an initial effort to both improve our website performance and minimize our calls to the API (in particular DarkSky). I think low-level caching is probably the way to go for this. 

- Memoizes creating a new Weather object in the GardenFacade and DashboardFacade
- Adds `Rails.cache.fetch` to the `get_weather` method in DarkSkyService
- Adds `Rails.cache.fetch` to the `get_location_data` method in LocationService
- Both caches take in instance-specific data(lat/long and zip code, respectively) so that it caches information specific to what we are looking for (instead of just returning the same weather data for all lat/longs)
- All tests passing
- Hits in test suite for `get_weather` API call down from 141 to 3
- Hits in test suite for `get_location_data` API call down from 14 to 2

### Caching resources: 
https://guides.rubyonrails.org/caching_with_rails.html#low-level-caching
https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html
https://devcenter.heroku.com/articles/caching-strategies

### CODE REVIEW

File: app/services/dark_sky_service.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 class DarkSkyService
 
   def get_weather(lat, long)
-    response = conn.get("/forecast/#{ENV["darksky_api_secret"]}/#{lat},#{long}")
-    JSON.parse(response.body, symbolize_names: true)
+    Rails.cache.fetch("weather_info_#{lat}_#{long}", expires_in: 1.hour) do
```
Comment (le3ah) [inline app/services/dark_sky_service.rb]: This is really great!

Comment (abenetka) [reply app/services/dark_sky_service.rb]: SO GREAT

File: app/services/location_service.rb [hunk 2]
```diff
@@ -14,10 +14,12 @@ def get_longitude
   private
   
   def get_location_data
-    response = conn.get("/maps/api/geocode/json") do |f|
-      f.params[:address] = @zip_code
+    Rails.cache.fetch("location_data_#{@zip_code}") do
```
Comment (abenetka) [inline app/services/location_service.rb]: So fancy and great

File: app/services/dark_sky_service.rb (hunk 1 above)
Comment (bendelonlee) [reply app/services/dark_sky_service.rb]: Great work!

**Pull request merged**